http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/AppComponent.scala

```diff
@@ -315,11 +315,13 @@ object appQuery extends TableQuery(new AppTable(_)) {
   def updateDateAccessed(appName: AppName, cloudContext: CloudContext, now: Instant): DBIO[Int] =
     cloudContext match {
       case CloudContext.Gcp(_) =>
-        // This query isn't efficient because appName isn't indexed. Future optimization is needed for findActiveByNameQuery method
-        findActiveByNameQuery(appName)
-          .filter(r => r.workspaceId.isEmpty)
```

Comment:
Why do you need a new query? Could you not filter by empty workspaces anymore?

Reply:
I could...it's just the query would be a bit less performant since `appName` is not indexed...and it's slightly less correct in that it's not filtering by `cloudContext` (altho think we're already doing this in a few other places, which is the reason why I didn't do the extra join in the original PR...figured I'll make things a bit more correct given this is relatively small PR itself)...

Reply:
That makes sense! For context this is the ticket that Lily worked on that made me pause https://broadworkbench.atlassian.net/browse/IA-4626

Reply:
1. The query itself: 
- Why do we need to join nodepool or kubernetes_cluster? It seems like all the fields we need are on APP.
- Do we need to preserve the original parts of `findActiveByNameQuery`? It appears that that search also includes these terms:
-- status not deleted 
-- no destroyed date

And why does this need to be a raw SQL query? It seems like it would be simple enough to write this as Slick.

All that said, I think the APP table is indexed by appName:
```
+-------+------------+--------------------+--------------+---------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| Table | Non_unique | Key_name           | Seq_in_index | Column_name   | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment |
+-------+------------+--------------------+--------------+---------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| APP   |          0 | PRIMARY            |            1 | id            | A         |           3 |     NULL | NULL   |      | BTREE      |         |               |
| APP   |          0 | IDX_APP_UNIQUE     |            1 | appName       | A         |           3 |     NULL | NULL   |      | BTREE      |         |               |
| APP   |          0 | IDX_APP_UNIQUE     |            2 | nodepoolId    | A         |           3 |     NULL | NULL   |      | BTREE      |         |               |
| APP   |          0 | IDX_APP_UNIQUE     |            3 | workspaceId   | A         |           3 |     NULL | NULL   | YES  | BTREE      |         |               |
| APP   |          0 | IDX_APP_UNIQUE     |            4 | destroyedDate | A         |           3 |     NULL | NULL   |      | BTREE      |         |               |
| APP   |          1 | FK_APP_NODEPOOL_ID |            1 | nodepoolId    | A         |           3 |     NULL | NULL   |      | BTREE      |         |               |
| APP   |          1 | FK_APP_DISK_ID     |            1 | diskId        | A         |           3 |     NULL | NULL   | YES  | BTREE      |         |               |
+-------+------------+--------------------+--------------+---------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
```
This is from my BEE; is prod different? Here it seems that a search on appName first would be indexed as expected.